lib/rouge/lexers/nasm.rb

```diff
@@ -18,84 +18,110 @@ class Nasm < RegexLexer
+           FNSTCW FNSTENV FNSTSW FPATAN FPREM FPREM1 FPTAN FRNDINT FRSTOR FSAVE
+           FSCALE FSETPM FSIN FSINCOS FSQRT FST FSTCW FSTENV FSTP FSTSW FSUB
+           FSUBP FSUBR FSUBRP FTST FUCOM FUCOMI FUCOMIP FUCOMP FUCOMPP FWAIT
+           FXAM FXCH FXTRACT FYL2X FYL2XP1 HLT IBTS ICEBP IDIV IMUL IN INC
+           INCBIN INSB INSD INSW INT INT01 INT03 INT1 INT3 INTO INVD INVLPG
+           INVLPGA INVPCID IRET IRETD IRETQ IRETW JCXZ JECXZ JMP JMPE JRCXZ LAHF
+           LAR LDS LEA LEAVE LES LFENCE LFS LGDT LGS LIDT LLDT LMSW LOADALL
+           LOADALL286 LODSB LODSD LODSQ LODSW LOOP LOOPE LOOPNE LOOPNZ LOOPZ LSL
+           LSS LTR MFENCE MONITOR MONITORX MOV MOVD MOVQ MOVSB MOVSD MOVSQ MOVSW
+           MOVSX MOVSXD MOVZX MUL MWAIT MWAITX NEG NOP NOT OR OUT OUTSB OUTSD
+           OUTSW PACKSSDW PACKSSWB PACKUSWB PADDB PADDD PADDSB PADDSIW PADDSW
+           PADDUSB PADDUSW PADDW PAND PANDN PAUSE PAVEB PAVGUSB PCMPEQB PCMPEQD
+           PCMPEQW PCMPGTB PCMPGTD PCMPGTW PDISTIB PF2ID PFACC PFADD PFCMPEQ
+           PFCMPGE PFCMPGT PFMAX PFMIN PFMUL PFRCP PFRCPIT1 PFRCPIT2 PFRSQIT1
+           PFRSQRT PFSUB PFSUBR PI2FD PMACHRIW PMADDWD PMAGW PMULHRIW PMULHRWA
+           PMULHRWC PMULHW PMULLW PMVGEZB PMVLZB PMVNZB PMVZB POP POPA POPAD
+           POPAW POPF POPFD POPFQ POPFW POR PREFETCH PREFETCHW PSLLD PSLLQ PSLLW
+           PSRAD PSRAW PSRLD PSRLQ PSRLW PSUBB PSUBD PSUBSB PSUBSIW PSUBSW
+           PSUBUSB PSUBUSW PSUBW PUNPCKHBW PUNPCKHDQ PUNPCKHWD PUNPCKLBW
+           PUNPCKLDQ PUNPCKLWD PUSH PUSHA PUSHAD PUSHAW PUSHF PUSHFD PUSHFQ
+           PUSHFW PXOR RCL RCR RDM RDMSR RDPMC RDSHR RDTSC RDTSCP RET RETF RETN
+           ROL ROR RSDC RSLDT RSM RSTS SAHF SAL SALC SAR SBB SCASB SCASD SCASQ
+           SCASW  SFENCE SGDT SHL SHLD SHR SHRD SIDT SKINIT SLDT SMI SMINT
+           SMINTOLD SMSW STC STD STI STOSB STOSD STOSQ STOSW STR SUB SVDC SVLDT
+           SVTS SWAPGS SYSCALL SYSENTER SYSEXIT SYSRET TEST UD0 UD1 UD2 UD2A
+           UD2B UMOV VERR VERW WBINVD WRMSR WRSHR XADD XBTS XCHG XLAT XLATB XOR
+
+           CMOVA CMOVAE CMOVB CMOVBE CMOVC CMOVE CMOVG CMOVGE CMOVL CMOVLE
+           CMOVNA CMOVNAE CMOVNB CMOVNBE CMOVNC CMOVNE CMOVNG CMOVNGE CMOVNL
+           CMOVNLE CMOVNO CMOVNP CMOVNS CMOVNZ CMOVO CMOVP CMOVPE CMOVPO CMOVS
+           CMOVZ
+
+           JA JAE JB JBE JC JCXZ JECXZ JE JG JGE JL JLE JNA JNAE JNB JNBE JNC
+           JNE JNG JNGE JNL JNLE JNO JNP JNS JNZ JO JP JPE JPO JS JZ
+
+           SETA SETAE SETB SETBE SETC SETE SETG SETGE SETL SETLE SETNA SETNAE
+           SETNB SETNBE SETNC SETNE SETNG SETNGE SETNL SETNLE SETNO SETNP SETNS
+           SETNZ SETO SETP SETPE SETPO SETS SETZ
         )
       end
 
       def self.keywords_type
         @keywords_type ||= Set.new %w(
-          DB DW DD DQ DT DO DY DZ RESB RESW RESD RESQ REST RESO RESY RESZ
-          db dq dd dq dt do dy dz resb resw resd resq rest reso resy resz
+          DB DW DD DQ DT DO DY DZ RESB RESW RESD RESQ REST RESO RESY RESZ db dq
```

Comment:
I think you meant dw (lowercase), otherwise we don't have it and get dq twice.

Reply:
No longer necessary with new lexer approach.